application/src/main/java/org/thingsboard/server/service/queue/DefaultTbCoreConsumerService.java

```diff
@@ -664,12 +665,9 @@ private void forwardToEventService(LifecycleEventProto eventProto, TbCallback ca
     }
 
     private void forwardToEventService(Event event, TbCallback callback) {
-        try {
-            actorContext.getEventService().saveAsync(event);
-            callback.onSuccess();
-        } catch (Exception e) {
-            callback.onFailure(e);
-        }
+        DonAsynchron.withCallback(actorContext.getEventService().saveAsync(event),
+                result -> callback.onSuccess(),
+                callback::onFailure);
```

Comment:
We need to use callabackExecutor. We can't process callbacks in the db call thread.